### PR TITLE
cp-355: Display only user chats

### PR DESCRIPTION
**Pull request by dmytro-kucherenko**
Done: 
- Only previously created chat is displayed on chat list after reloading the chat page.

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92298163/1f312e20-23b6-4858-acd3-2ad31f5fec9a

### CODE REVIEW

**Review by juanc-escobar — DISMISSED** [review]
everything looks good, great job @Dmytro-Kucherenko  ✅

File: backend/src/packages/chats/chat.repository.ts [hunk 1]
```diff
@@ -25,8 +32,15 @@ class ChatRepository implements Repository {
     const chats = await this.chatModel
       .query()
       .withGraphJoined(ChatsRelation.MEMBERS)
-      .whereExists(UserToChatModel.query().where('userId', userId))
-      .orderBy('createdAt', 'DESC')
+      .whereExists(
+        this.userToChatModel
+          .query()
+          .whereRaw(
+            `${this.userToChatModel.tableName}.chat_id = ${this.chatModel.tableName}.id`,
```
Comment (v0ldemar01) [inline backend/src/packages/chats/chat.repository.ts]: let's use relatedQuery here

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: backend/src/packages/chats/chat.repository.ts (hunk 1 above)
Comment (dmytro-kucherenko) [reply backend/src/packages/chats/chat.repository.ts]: Done

File: backend/src/packages/chats/chat.repository.ts [hunk 2]
```diff
@@ -29,17 +29,10 @@ class ChatRepository implements Repository {
   }
 
   public async findAllByUserId(userId: number): Promise<ChatEntity[]> {
-    const chats = await this.chatModel
-      .query()
+    const chats = await this.userToChatModel
+      .relatedQuery(UserToChatRelation.CHAT)
+      .for(this.userToChatModel.query().where('userId', userId))
```
Comment (v0ldemar01) [inline backend/src/packages/chats/chat.repository.ts]:
```suggestion
      .for(this.userToChatModel.query().where({ userId }))
```

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]
Just one improvement)

File: backend/src/packages/chats/chat.repository.ts (hunk 2 above)
Comment (dmytro-kucherenko) [reply backend/src/packages/chats/chat.repository.ts]: Done

**Review by v0ldemar01 — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**